modules/react/common/lib/utils/useModalityType.ts

```diff
@@ -11,24 +11,24 @@ type Modality = 'mouse' | 'touch' | 'pen';
 //   a. if localStorage isn't primed, use clientWidth of the browsers
 //   b. if < 768, default to 'touch'
 //   c. else default to 'mouse'
-const localStorageValue = 'mouse' as Modality; //((typeof localStorage !== 'undefined'
-//   ? localStorage.getItem('modality')
-//   : '') ||
-//   (typeof document !== 'undefined'
-//     ? document.documentElement.clientWidth < 768
-//       ? 'touch'
-//       : ''
-//     : '') ||
-//   'mouse') as Modality;
+let localStorageValue = ((typeof localStorage !== 'undefined'
+  ? localStorage.getItem('modality')
+  : '') ||
+  (typeof document !== 'undefined'
+    ? document.documentElement.clientWidth < 768
+      ? 'touch'
+      : ''
+    : '') ||
+  'mouse') as Modality;
```

Comment:
This was all commented out so just uncommented it.

Reply:
This logic looks huge and there is a lot of repeats elements. Can we switch it to something like this:

```
const localStorageValue =
  ((localStorage && localStorage.getItem('modality')) as Modality) ||
  (document && document.documentElement.clientWidth < 768 && 'touch') ||
  'mouse';

```
So it validates if localStorage has some values (it covers all false possibilities) to try to get 'modality' value (and this is only value type that should be defined instead of the whole block)
if it's not it validates document and document client width. Otherwise it uses 'mouse'.

Reply:
works for me, nice clean up!